src/grpc/dp_grpc_impl.c

```diff
@@ -393,20 +397,28 @@ static int dp_process_create_prefix(struct dp_grpc_responder *responder)
 	if (!port)
 		return DP_GRPC_ERR_NO_VM;
 
+	iface_vni = port->iface.vni;
 	if (request->addr.ip_type == RTE_ETHER_TYPE_IPV4) {
-		iface_vni = port->iface.vni;
 		ret = dp_add_route(port, iface_vni, 0, request->addr.ipv4, NULL, request->length);
 		if (DP_FAILED(ret))
 			return ret;
 
-		if (DP_FAILED(dp_create_vnf_route(ul_addr6, DP_VNF_TYPE_ALIAS_PFX, iface_vni, port, request->addr.ipv4, request->length))) {
+		if (DP_FAILED(dp_create_vnf_route(ul_addr6, DP_VNF_TYPE_ALIAS_PFX, iface_vni, port, &request->addr, request->length))) {
 			dp_del_route(port, iface_vni, request->addr.ipv4, request->length);
 			return DP_GRPC_ERR_VNF_INSERT;
 		}
-		rte_memcpy(reply->addr6, ul_addr6, sizeof(reply->addr6));
+	} else if (request->addr.ip_type == RTE_ETHER_TYPE_IPV6) {
+		ret = dp_add_route6(port, iface_vni, 0, request->addr.ipv6, NULL, request->length);
+		if (DP_FAILED(ret))
+			return ret;
+
+		if (DP_FAILED(dp_create_vnf_route(ul_addr6, DP_VNF_TYPE_ALIAS_PFX, iface_vni, port, &request->addr, request->length))) {
+			dp_del_route6(port, iface_vni, request->addr.ipv6, request->length);
+			return DP_GRPC_ERR_VNF_INSERT;
+		}
```

Comment:
Since you already do that for `dp_create_vnf_route` wouldn't it be beneficial for `dp_add_route` and `dp_del_route` to actually accept `struct dp_ip_address`?
That way this code would be much simpler (and there is the same effect in `dp_process_create_route`.

Reply:
I would agree but this is more of a refactoring which we can do in a follow-up refactoring PR. This would bloat the diff here as well.